airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/common/sql/sql_processor.py

```diff
@@ -0,0 +1,979 @@
+        self._execute_sql(
+            text(
+                f"ALTER TABLE {self._fully_qualified(table.name)} "
+                f"ADD COLUMN {column_name} {column_type}"
+            ),
+        )
+
+    def _add_missing_columns_to_table(
+        self,
+        stream_name: str,
+        table_name: str,
+    ) -> None:
+        """Add missing columns to the table.
+
+        This is a no-op if all columns are already present.
+        """
+        columns = self._get_sql_column_definitions(stream_name)
+        # First check without forcing a refresh of the cache (faster). If nothing is missing,
+        # then we're done.
+        table = self._get_table_by_name(
+            table_name,
+            force_refresh=False,
+        )
+        missing_columns: bool = any(column_name not in table.columns for column_name in columns)
+
+        if missing_columns:
+            # If we found missing columns, refresh the cache and then take action on anything
+            # that's still confirmed missing.
+            columns_added = False
+            table = self._get_table_by_name(
+                table_name,
+                force_refresh=True,
+            )
+            for column_name, column_type in columns.items():
+                if column_name not in table.columns:
+                    self._add_column_to_table(table, column_name, column_type)
+                    columns_added = True
+
+            if columns_added:
+                # We've added columns, so invalidate the cache.
+                self._invalidate_table_cache(table_name)
+
+    @final
+    def _write_temp_table_to_final_table(
+        self,
+        stream_name: str,
+        temp_table_name: str,
+        final_table_name: str,
+        write_strategy: WriteStrategy,
+    ) -> None:
+        """Write the temp table into the final table using the provided write strategy."""
+        has_pks: bool = bool(self._get_primary_keys(stream_name))
+        has_incremental_key: bool = bool(self._get_incremental_key(stream_name))
+        if write_strategy == WriteStrategy.MERGE and not has_pks:
+            raise exc.PyAirbyteInputError(
+                message="Cannot use merge strategy on a stream with no primary keys.",
+                context={
+                    "stream_name": stream_name,
+                },
+            )
+
+        if write_strategy == WriteStrategy.AUTO:
+            configured_destination_sync_mode: DestinationSyncMode = (
+                self.catalog_provider.get_destination_sync_mode(stream_name)
+            )
+            if configured_destination_sync_mode == DestinationSyncMode.overwrite:
+                write_strategy = WriteStrategy.REPLACE
+            elif configured_destination_sync_mode == DestinationSyncMode.append:
+                write_strategy = WriteStrategy.APPEND
+            elif configured_destination_sync_mode == DestinationSyncMode.append_dedup:
+                write_strategy = WriteStrategy.MERGE
+
```

Comment:
This is a new addition to the `SqlProcessor` logic. Now, `WriteStrategy.AUTO` will defer to the configured catalog's "destination sync mode".